ranger-companion.lic

```diff
@@ -30,56 +29,55 @@ class Companion
-        bput('pet raccoon', 'You pet', 'Touch what')
+        DRC.bput('pet raccoon', 'You pet', 'Touch what')
       end
       if line =~ /^A .* wolf begins to whimper./
-        pause 1 until pause_all
+        pause 1 until DRC.pause_all
         waitrt?
-        bput('stow left', 'Stow what', 'You put')
-        case bput('get my milk', 'You get', 'You are already holding', 'What were you referring to')
+        DRC.bput('stow left', 'Stow what', 'You put')
+        case DRC.bput('get my milk', 'You get', 'You are already holding', 'What were you referring to')
         when 'What were you referring to'
-          bput('signal companion to sleep', 'wolf wanders off to find', 'Your companion is not', 'You have no companion', 'snapping your fingers')
+          DRC.bput('signal companion to sleep', 'wolf wanders off to find', 'Your companion is not', 'You have no companion', 'snapping your fingers')
           exit
         end
-        case bput('feed my milk to wolf', 'The baby wolf greedily drinks', "It doesn't seem hungry", "You can't feed with", 'wolf shies away from you')
+        case DRC.bput('feed my milk to wolf', 'The baby wolf greedily drinks', "It doesn't seem hungry", "You can't feed with", 'wolf shies away from you')
         when 'wolf shies away from you'
-          bput('feed my milk to second wolf', 'The baby wolf greedily drinks', "It doesn't seem hungry", "You can't feed with", 'wolf shies away from you')
+          DRC.bput('feed my milk to second wolf', 'The baby wolf greedily drinks', "It doesn't seem hungry", "You can't feed with", 'wolf shies away from you')
         when "You can't feed with"
-          bput('signal companion to sleep', 'wolf wanders off to find', 'Your companion is not', 'You have no companion', 'snapping your fingers')
+          DRC.bput('signal companion to sleep', 'wolf wanders off to find', 'Your companion is not', 'You have no companion', 'snapping your fingers')
           exit
         end
-        bput('stow my milk', 'You put your', 'Stow what')
-        unpause_all
+        DRC.bput('stow my milk', 'You put your', 'Stow what')
       end
       next unless line =~ /^A .* raccoon begins to whimper./
-      pause 1 until pause_all
+      pause 1 until DRC.pause_all
       waitrt?
-      bput('stow left', 'Stow what', 'You put')
-      case bput('get my corn', 'You get', 'You are already holding', 'What were you referring to')
+      DRC.bput('stow left', 'Stow what', 'You put')
+      case DRC.bput('get my corn', 'You get', 'You are already holding', 'What were you referring to')
       when 'What were you referring to'
-        bput('signal companion to sleep', 'raccoon wanders off to find a quiet place to sleep.')
+        DRC.bput('signal companion to sleep', 'raccoon wanders off to find a quiet place to sleep.')
         exit
       end
-      bput('feed my corn to raccoon', 'raccoon greedily eats')
-      bput('stow my corn', 'You put your', 'Stow what')
-      unpause_all
```

Comment:
This particular `unpause_all` is occurring inside a loop (starts at line 23 and ends at line 76). Can we keep this here so that while the code is going through the "happy path" it'll unpause each iteration and not pause again until conditions are met?

The `before_dying` is of course a great idea to handle the myriad of places this script might choose to `exit` prematuerly

Reply:
Man, that is a great catch/point. You and Robert are great reviewers. Fixed with latest commit.

Reply:
Commenting on your reply edit - the script is designed to continue to run while your companion is out, and take action of various sorts to keep your companion happy. I think you're exactly right on this loop. We need to unpause_all at the end of this loop. The only known issue with the script was when it exited out of the loop and failed to unpause_all.